src/core/symbol/SymbolIdGenerator.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 SYMBOL
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"),
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { toBigIntLE, toBufferLE } from 'bigint-buffer';
+import * as Crypto from 'crypto';
+import { sha3_256 } from 'js-sha3';
+import { NamespaceConst } from '../Constants';
+import { Converter } from '../utils/Converter';
+
+export class SymbolIdGenerator {
+    /**
+     * Generate a mosaic id
+     * @param {Uint8Array} ownerAddress Mosaic owner's address bytes
+     * @param {Uint8Array} nonce Mosaic nonce bytes
+     * @returns {BigInt}
+     */
+    public static generateMosaicId(ownerAddress: Uint8Array, nonce: Uint8Array): bigint {
```

Comment:
outside the scope of this review, but you might want to consider having a distinct type for address instead of just `Uint8Array`